kpops/components/base_components/pipeline_component.py

```diff
@@ -154,14 +154,16 @@ def apply_from_inputs(self, name: str, topic: FromTopic) -> None:
         :type topic: FromTopic
         """
         match topic.type:
+            case None:
+                self.add_extra_input_topic(
+                    topic.role, [name]  # pyright: ignore [reportGeneralTypeIssues]
```

Comment:
don't we need to check if `topic.role` is set?

Reply:
It's covered in the `from` [validator](https://github.com/bakdata/kpops/blob/refactor/input-output-types/kpops/components/base_components/models/from_section.py#L45-L47)

Now that I see it, perhaps the validator doesn't need to assign a value to `type` and instead should only validate?
Same happens with the `to` section.

Reply:
yeah, I guess you could technically skip the assignment if you're covering `None` here anyways.